src/operator/roi_pooling.cc

```diff
@@ -227,15 +227,49 @@ Operator *ROIPoolingProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_
 DMLC_REGISTER_PARAMETER(ROIPoolingParam);
 
 MXNET_REGISTER_OP_PROPERTY(ROIPooling, ROIPoolingProp)
-.describe("Performs region-of-interest pooling on inputs. Resize bounding box coordinates by "
-"spatial_scale and crop input feature maps accordingly. The cropped feature maps are pooled "
-"by max pooling to a fixed size output indicated by pooled_size. batch_size will change to "
-"the number of region bounding boxes after ROIPooling")
-.add_argument("data", "NDArray-or-Symbol", "Input data to the pooling operator, a 4D Feature maps")
+.describe(R"code(Performs region of interest(ROI) pooling on the input array.
+
+ROI pooling is a variant of max pooling layer, in which output size is fixed and
+input rectangle is a parameter. It is a neural-net layer mostly used in training
+a Fast R-CNN network for object detection tasks.
+
+This operator takes a 4D feature map as input array and rectangular windows into a conv feature map
+as rois and produces a fixed sized output array regardless of the ROI size.
+
+To crop input feature map according to your interest, resize the bounding box coordinates
+by changing parameters `rois` and `spatial_scale`.
+
+The cropped feature maps are pooled by standard max pooling to a fixed size output indicated by
+`pooled_size` parameter. batch_size will change to the number of region bounding boxes after ROIPooling.
+
+Examples::
+
+  x = [[[[  0.,   1.,   2.,   3.],
+         [  4.,   5.,   6.,   7.],
+         [  8.,   9.,  10.,  11.],
+         [ 12.,  13.,  14.,  15.]],
+
+        [[ 16.,  17.,  18.,  19.],
+         [ 20.,  21.,  22.,  23.],
+         [ 24.,  25.,  26.,  27.],
+         [ 28.,  29.,  30.,  31.]]]
+
+  // roi i.e. bounding box coordinates
+  y = [[0,0,0,3,3]]
+
+  // returns array of shape (2,2) according to the given roi with max pooling
+  ROIPooling(x, y, (2,2), 1.0) = [[[[  5.,   7.],
+                                    [ 13.,  15.]],
+
+                                   [[ 21.,  23.],
+                                    [ 29.,  31.]]]]
+
+)code" ADD_FILELINE)
+.add_argument("data", "NDArray-or-Symbol", "The input array to the pooling operator, a 4D Feature maps")
```

Comment:
Feature maps might not be an appropriate term to use here, Activation map? @mli ?